docs/pages/architecture/agents.mdx

```diff
@@ -0,0 +1,265 @@
+- The **cluster name**. On cloud-hosted Teleport Enterprise clusters, this is a
+  subdomain of `teleport.sh`. On self-hosted clusters, the default value is the
+  `hostname` of the Teleport Auth Service, and users can configure this.
+- The agent's **role** (e.g., `node,proxy`) encoded as a certificate extension. 
+- The **expiry time** of the certificate.
+
+### Credentials for Teleport clients
+
+When a user logs into the cluster with the `tsh login` command, the Teleport
+Auth Service uses the `user` certificate authority to sign the certificate. See
+[Issuing User Certificates](../architecture/authentication.mdx) for more details
+on how it works. If the user authenticates via the Teleport Web UI, the browser
+retrieves a session cookie.
+
+In most cases, Teleport users must retrieve another user certificate in order to
+access Teleport-protected resources. The certificate authorizes the user to
+access specific resources, such as an application or database. The following
+table indicates how a user obtains this certificate for different kinds of
+resources:
+
+|Resource kind|How users obtain certificates|
+|---|---|
+|Databases|The user can retrieve a client certificate for before connecting to a database with `tsh db login`. The `tsh db connect` and `tsh proxy db` commands also retrieve certificates. The Teleport Database Service verifies the certificate when a user connects to a database.|
+|Kubernetes clusters|The user runs `tsh kube login`, which updates the local kubeconfig to contain a certificate signed by the Teleport user CA. The user can then execute `kubectl` commands against Teleport-protected Kubernetes clusters.|
+|Servers|After running `tsh login`, the user receives a certificate that contains authorization information for servers. The SSH Service checks the certificate when a user authenticates to a Teleport-protected server.|
+|Web applications|The user authenticates to the Teleport Web UI and visits a Teleport-protected application. The Teleport Proxy Service forwards traffic to the application along with Teleport-signed JSON web tokens that the application can verify. For HTTP API applications, TCP applications, and cloud provider APIs, the user runs `tsh apps login` to retrieve a certificate.|
+|Windows desktops|Users authenticate to the Teleport Web UI. The Web UI initiates a WebSocket session with the Teleport Desktop Service and communicates with the service using Teleport Desktop Protocol (TDP). The Desktop Service converts TDP traffic to Remote Desktop Protocol traffic.|
+
+In most cases (except for some self-hosted clusters), the user will have
+connected to the Auth Service through the Teleport Proxy Service. The Auth
+Service forwards user certificates through the Proxy Service over mutual TLS.  
+
+## Establishing reverse tunnels
+
+In most cases, when an agent joins a Teleport cluster, it establishes a reverse
+SSH tunnel to the Proxy Service. As such, users do not need to have direct
```

Comment:
nit: standard nomenclature is "SSH reverse tunnel" rather than "reverse SSH tunnel"
```suggestion
In most cases, when an agent joins a Teleport cluster, it establishes an SSH
reverse tunnel to the Proxy Service. As such, users do not need to have direct
```